boards/DIM/Test/Src/seven_seg_displays_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include "fff.h"
+#include "gtest/gtest.h"
+
+extern "C"
+{
+#include "App_SevenSegDisplay.h"
+#include "App_SevenSegDisplays.h"
+
+    DEFINE_FFF_GLOBALS;
+    FAKE_VALUE_FUNC(float, get_state_of_charge);
+    FAKE_VOID_FUNC(set_right_hex_digit, uint8_t);
+    FAKE_VOID_FUNC(set_middle_hex_digit, uint8_t);
+    FAKE_VOID_FUNC(set_left_hex_digit, uint8_t);
+}
+
+class SevenSegDisplaysTest : public testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        right_display  = App_SevenSegDisplay_Create(set_right_hex_digit);
+        middle_display = App_SevenSegDisplay_Create(set_middle_hex_digit);
+        left_display   = App_SevenSegDisplay_Create(set_left_hex_digit);
+
+        seven_segment_displays = App_SevenSegDisplays_Create(
+            get_state_of_charge, left_display, middle_display, right_display);
+
+        RESET_FAKE(get_state_of_charge);
+        RESET_FAKE(set_right_hex_digit);
+        RESET_FAKE(set_middle_hex_digit);
+        RESET_FAKE(set_left_hex_digit);
+
+        FFF_RESET_HISTORY();
+    }
+    void TearDown() override
+    {
+        App_SevenSegDisplay_Destroy(right_display);
+        App_SevenSegDisplay_Destroy(middle_display);
+        App_SevenSegDisplay_Destroy(left_display);
+        App_SevenSegDisplays_Destroy(seven_segment_displays);
+    }
+
+    struct SevenSegDisplay * left_display;
+    struct SevenSegDisplay * middle_display;
+    struct SevenSegDisplay * right_display;
+    struct SevenSegDisplays *seven_segment_displays;
+};
+
+TEST_F(SevenSegDisplaysTest, write_max_state_of_chrage)
```

Comment:
Happy to add more unit tests. But can't think of any more things to test for now. Suggestions are welcomed.